src/main/java/io/cdap/e2e/pages/actions/CdfSignInActions.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.e2e.pages.actions;
+
+import io.cdap.e2e.pages.locators.CdfSignInLocator;
+import io.cdap.e2e.utils.ConstantsUtil;
+import io.cdap.e2e.utils.ElementHelper;
+import io.cdap.e2e.utils.SeleniumHelper;
+import io.cdap.e2e.utils.WaitHelper;
+import java.io.IOException;
+
+
+
+/**
+ * Represents CdfSignInActions
+ */
+public class CdfSignInActions {
+    private static CdfSignInLocator cdfSignInLocator;
+    static {
+        cdfSignInLocator = SeleniumHelper.getPropertiesLocators(CdfSignInLocator.class);
+    }
+
+    public static void login() throws IOException {
+        ElementHelper.sendKeys(cdfSignInLocator.cdfUsername, SeleniumHelper.readParameters(ConstantsUtil.CDFUSERNAME));
+        ElementHelper.clickOnElement(cdfSignInLocator.nextButton);
+        ElementHelper.sendKeys(cdfSignInLocator.cdfPassword, SeleniumHelper.readParameters(ConstantsUtil.CDFPASSWORD));
+        ElementHelper.clickOnElement(cdfSignInLocator.nextButton);
+
+        // Adding optional operations here.
+
+        /*
+        Based on the observations in case of CDF test execution from local we encounter Allow Button
+        and in case of CDF test execution from Cloud Build we encounter Continue Button that's why we
+        are using 2 confirmation buttons here.
+        */
+
+        String testAccountName = SeleniumHelper.readParameters("testAccountName");
+        ElementHelper.clickIfDisplayed(cdfSignInLocator.selectTestAccount(testAccountName),
+          ConstantsUtil.MEDIUM_TIMEOUT_SECONDS, cdfSignInLocator.clickOnAllowButton());
+
+        ElementHelper.clickIfDisplayed(cdfSignInLocator.clickOnContinueButton(), ConstantsUtil.MEDIUM_TIMEOUT_SECONDS,
+          cdfSignInLocator.locatePluginNameInList(ConstantsUtil.FIRST_PLUGIN_IN_LIST, "Source"));
+
+        ElementHelper.clickIfDisplayed(cdfSignInLocator.clickOnAllowButton(), ConstantsUtil.MEDIUM_TIMEOUT_SECONDS,
+          cdfSignInLocator.locatePluginNameInList(ConstantsUtil.FIRST_PLUGIN_IN_LIST, "Source"));
```

Comment:
Why do we want to combine two operations here, why can't we do something like this?
1. Click on `element1` if displayed
2. Click on `element2` if displayed
3. Wait until `plugin element` is displayed.

Reply:
The main reason behind creating ```clickIfDisplayed``` was to implement a retry if the element is not displayed on the first attempt. 

The reason to combine 2 operations was related to this retry only, the idea was that if once we clicked on the desired element we will validate if the next expected element (3rd parameter) is displayed if we don't see the next element we try to click on the desired element again.

Reply:
Will this retry be implemented in future? I don't see it being retried in this PR or am I missing something?

Reply:
The Retry logic is actually implemented in ```clickIfDisplayed``` 

```

    clickIfDisplayed(locatorToClick, timeOutInSeconds);

    // If the next webElement is not displayed we will click on the desired element again.
    if (!isElementDisplayed(locatorToWaitFor, timeOutInSeconds)) {
    logger.info("Retry : Click on " + locatorToClick);
    clickIfDisplayed(locatorToClick);
      
```


(https://github.com/cdapio/cdap-e2e-tests/pull/197/files#diff-1b63d2aab0be21d4b99280309e329d962acb6a5c2bb82e6075c8da4ce3b6569eR134)

Reply:
why is the retry needed? The click should result in the next element within timeout. If we retry, we might initiate the same operation twice.

Reply:
I agree that this can result in same operation twice, the main purpose of adding the refresh here was to avoid any flakiness in case of CDF login flow because I observed few failed executions while executing the flow. I am not completely sure why but sometimes the click was getting intercepted. (We were waiting for elementToBeClickable still there were some failed executions due to this)

We can remove this refresh and use a simple click with a timeout, I will just have to make sure that there is no flakiness while using a simple click.

what do you guys suggest ? (cc : @itsankit-google)